aws/resource_aws_elasticsearch_domain.go

```diff
@@ -158,6 +158,16 @@ func resourceAwsElasticSearchDomainImport(
 func resourceAwsElasticSearchDomainCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).esconn
 
+	// The API doesn't check for duplicate names
+	// so w/out this check Create would act as upsert
+	// and might cause duplicate domain to appear in state
+	resp, err := conn.DescribeElasticsearchDomain(&elasticsearch.DescribeElasticsearchDomainInput{
+		DomainName: aws.String(d.Get("domain_name").(string)),
+	})
```

Comment:
Would just add a log here, to get better debug details

Reply:
Actually I think that's unnecessary, because we already have full error response in its original format (as any other HTTP response coming from the AWS API)?